accord-core/src/main/java/accord/messages/PreAccept.java

```diff
@@ -110,26 +111,32 @@ public PreAcceptReply apply(SafeCommandStore safeStore)
         // note: this diverges from the paper, in that instead of waiting for JoinShard,
         //       we PreAccept to both old and new topologies and require quorums in both.
         //       This necessitates sending to ALL replicas of old topology, not only electorate (as fast path may be unreachable).
-        if (minUnsyncedEpoch < txnId.epoch() && !safeStore.ranges().coordinates(txnId).intersects(scope))
+        if (minEpoch < txnId.epoch() && !safeStore.ranges().coordinates(txnId).intersects(scope))
             return applyIfDoesNotCoordinate(safeStore);
 
         SafeCommand safeCommand = safeStore.get(txnId, this, route);
-        switch (Commands.preaccept(safeStore, safeCommand, txnId, maxEpoch, partialTxn, route, progressKey))
+        Commands.AcceptOutcome outcome = Commands.preaccept(safeStore, safeCommand, txnId, maxEpoch, partialTxn, route);
+        Command command = safeCommand.current();
+        switch (outcome)
         {
             default:
-            case Success:
                 // we might hit 'Redundant' if we have to contact later epochs and partially re-contact a node we already contacted
-                // TODO (expected): consider dedicated special case, or rename
+                // TODO (desired): consider dedicated special case, or rename
             case Redundant:
-                Command command = safeCommand.current();
-                // for efficiency, we don't usually return dependencies newer than txnId as they aren't necessarily needed
-                // for recovery, and it's better to persist less data than more. However, for exclusive sync points we
-                // don't need to perform an Accept round, nor do we need to persist this state to aid recovery. We just
-                // want the issuer of the sync point to know which transactions to wait for before it can safely treat
-                // all transactions with lower txnId as expired.
-                Ranges ranges = safeStore.ranges().allBetween(minUnsyncedEpoch, txnId);
-                return new PreAcceptOk(txnId, command.executeAt(),
-                                       calculatePartialDeps(safeStore, txnId, command.partialTxn().keys(), scope, EpochSupplier.constant(minUnsyncedEpoch), txnId, ranges));
+            case Success:
+                if (command.saveStatus().compareTo(SaveStatus.PreAccepted) > 0)
```

Comment:
Either the coordinator has already moved on to Accept and later phases, so this has been delivered out of order and the coordinator will discard this nack, or it has already been interrupted and the command is being successfully recovered by some recovery coordinator in which case it should stop.

Reply:
To be more specific, this is to avoid a case where we try to propose bad dependencies and trigger an invariant failure, that is otherwise harmless as the deps would be rejected. But, we should anyway not be attempting to propose an `Accept` if we have encountered an `Accept` or later during our `PreAccept` phase.